projects/js-packages/components/components/threat-modal/user-connection-gate.tsx

```diff
@@ -0,0 +1,65 @@
+import { Text, Button } from '@automattic/jetpack-components';
+import { Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { ReactNode } from 'react';
+import styles from './styles.module.scss';
+
+const UserConnectionGate = ( {
+	userConnectionNeeded,
+	userIsConnecting,
+	handleConnectUser,
+	children,
+}: {
+	userConnectionNeeded: boolean;
+	userIsConnecting: boolean;
+	handleConnectUser: () => void;
+	children: ReactNode;
+} ) => {
+	if ( userConnectionNeeded ) {
+		return (
+			<>
+				<Notice
+					status="warning"
+					isDismissible={ false }
+					children={
+						<Text>
+							{ __(
+								'Before Jetpack can ignore and auto-fix threats on your site, a user connection is needed.',
+								'jetpack'
+							) }
+						</Text>
+					}
+				/>
+
+				<Text>
+					{ __(
+						'A user connection provides Jetpack the access necessary to perform these tasks.',
+						'jetpack'
+					) }
+				</Text>
+
+				<Text>
+					{ __(
+						'Once you’ve secured a user connection, all Jetpack features will be available for use.',
+						'jetpack'
+					) }
+				</Text>
+
+				<div className={ styles[ 'modal-actions' ] }>
+					<Button
+						isExternalLink={ true }
+						weight="regular"
+						isLoading={ userIsConnecting }
+						onClick={ handleConnectUser }
+					>
+						{ __( 'Connect your user account', 'jetpack' ) }
+					</Button>
+				</div>
+			</>
+		);
+	}
+
+	return <>{ children }</>;
```

Comment:
```suggestion
	return children;
```
Double nit - Can just return `children`, this could also be done as an early return at the top (personal preference).

Reply:
Fixed in [113862a](https://github.com/Automattic/jetpack/pull/40204/commits/113862ae8d6e51cc0ac3349f1b27e2e1878da651)